File: elixir/factorial.exs
n = IO.gets("Enter a number: ") |> String.trim() |> String.to_integer()

cond do
  n > 0 ->
    fact_n = Enum.reduce(1..n, 1, &(&1 * &2))

    IO.puts("#{n}! is equal to #{fact_n}")

  n == 0 ->
    IO.puts("0! is equal to 1")

  n < 0 ->
    IO.puts("#{n}! does not exist!")
end

File: Python/Factorial.py

num = int(input("Enter a number: "))
factorial = 1
if num < 0:
num = int(input("Enter a number: "))                       #Taken number from user
factorial = 1                                              #defining variable
if num < 0:                                                #checking number is negative or not
   print("Sorry, factorial does not exist for negative numbers")
elif num == 0:
elif num == 0:                                             #checking number is zero or not
   print("The factorial of 0 is 1")
else:
   for i in range(1,num + 1):
   for i in range(1,num + 1):                               #using for loop this program is done
       factorial = factorial*i
   print("The factorial of",num,"is",factorial)
